internal/index_read.go

```diff
@@ -165,15 +159,24 @@ func readIntervals(r io.Reader, typ string) ([]bgzf.Offset, error) {
 	if n == 0 {
 		return nil, nil
 	}
-	var vOff uint64
 	offsets := make([]bgzf.Offset, n)
-	for i := range offsets {
-		err := binary.Read(r, binary.LittleEndian, &vOff)
+	// chunkSize determines the number of offsets consumed by each binary.Read.
+	chunkSize := 512
+	vOffs := make([]uint64, chunkSize)
+	for i := 0; i < int(n); i += chunkSize {
+		// Limit the size of vOffs on final iteration.
+		if i+chunkSize >= int(n) {
+			vOffs = vOffs[:int(n)-i]
+		}
+		err := binary.Read(r, binary.LittleEndian, &vOffs)
```

Comment:
`err = binary.Read(r, binary.LittleEndian, vOffs[:min(int(n)-1, chunkSize)])` and define `min(int, int) int`\*. Then the condition above goes away, you can use the `[chunkSize]uint64` value and readability improves.

Note that because `vOffs` is a slice in the existing code and the proposed code here, the address operator is not required.

\*  `min(int, int) int`
```
func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}
```